tests/testthat/test-git.R

```diff
@@ -44,20 +44,31 @@ test_that("git user and email can be retrieved", {
   expect_equal(email, "email@example.com")
 })
 
+test_that("git config with invalid field", {
+  path <- build_git_demo()
+  expect_error(git_config(path, "favourite_vegetable"),
+               "git config field not found")
+})
+
 test_that("git branch can be retrieved", {
   path <- build_git_demo()
   branch <- git_branch(path)
   expect_equal(branch, "master")
 
+# Not sure how to detach head with gert, so will
```

Comment:
I think `git_info` has the information we need to work this out. Doing an info with a detached head we get

```
$path
[1] "/home/rashton/projects/dettl/"

$head
[1] "HEAD"

$shorthand
[1] "HEAD"

$commit
[1] "724dc885afe37a79caeb64bde3a742911a317f07"

$remote
[1] NA

$upstream
[1] NA

...
```
Whereas when we're on a branch we have 
```
$head
[1] "refs/heads/vimc-3004"
...
```
I think probably enough to check that `$head` points to an actual branch i.e. `git_info()$head %in% git_info()$reflist`

What do you think?

Reply:
Fixed with mockery!